lgc/patch/ShaderInputs.cpp

```diff
@@ -343,9 +348,24 @@ void ShaderInputs::fixupUses(Module &module, PipelineState *pipelineState) {
           continue;
       }
 
-      value->setName(getInputName(static_cast<ShaderInput>(kind)));
+      auto inputKind = static_cast<ShaderInput>(kind);
+      value->setName(getInputName(inputKind));
+      const bool needCheckWorgroupId = (inputKind == ShaderInput::WorkgroupId) && (stage == ShaderStageCompute);
       for (Instruction *&call : inputUsage->users) {
         if (call && call->getFunction() == &func) {
+          if (needCheckWorgroupId) {
+            for (auto user : call->users()) {
+              if (auto extractInst = dyn_cast<ExtractElementInst>(user)) {
+                if (auto indexInst = dyn_cast<ConstantInt>(extractInst->getIndexOperand())) {
+                  unsigned index = indexInst->getZExtValue();
+                  assert(index < 3);
+                  useWorkgroupIds[index] = true;
+                }
+              } else {
+                useWholeWorkgroupId = true;
+              }
```

Comment:
I believe useWholeWorkgroupId also needs to be set in the case of `extractelement` with dynamic index.
```suggestion
              if (auto extractInst = dyn_cast<ExtractElementInst>(user)) {
                if (auto indexInst = dyn_cast<ConstantInt>(extractInst->getIndexOperand())) {
                  unsigned index = indexInst->getZExtValue();
                  assert(index < 3);
                  useWorkgroupIds[index] = true;
                  continue;
                }
              }
              
              useWholeWorkgroupId = true;
```

Reply:
Thank you for catching it. Fixed.